circuit_knitting/cutting/cutting_evaluation.py

```diff
@@ -323,25 +316,6 @@ def _generate_cutting_experiments(
                 meas_qc = _append_measurement_circuit(decomp_qc, cog)
                 subexperiments_dict[label].append(meas_qc)
 
-    # Generate legacy subexperiments list
```

Comment:
We actually don't need the legacy list anymore because we will just pass the outputs of Sampler directly out of `execute_experiments`. The 3D list structure will be removed entirely in this PR